### PR TITLE
feat(stats-detectors): Add regression group model

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds a regression group model for tracking metadata about a regression.

### CODE REVIEW

File: src/sentry/models/statistical_detectors.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    active = models.BooleanField(default=True)
+
+    project_id = BoundedBigIntegerField(db_index=True)
+
+    type = BoundedIntegerField(choices=RegressionType.as_choices())
+
+    # The fingerprint sent to the issue platform which
+    # accepts a list of strings. This corresponds to the
+    # first string which is a 8-16 char hex value.
+    # Here, it's being stored as it's integer equivalent.
```
Comment (markstory) [inline src/sentry/models/statistical_detectors.py:59]: Why as an integer?

Comment (Zylphrex) [reply src/sentry/models/statistical_detectors.py:59]: I don't have a strong preference here to store it as an integer or string. Just thought it'd be more efficient to store it as an integer as I'll be filtering on this column quite a bit.

If storing it as a string is preferred, happy to switch it.

Comment (markstory) [reply src/sentry/models/statistical_detectors.py]: Strings result in fewer translations needed and unless this is going to be 10s of billions of rows the storage delta isn't likely to matter much.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**